src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![feature(c_variadic, ptr_wrapping_offset_from)]
+#![deny(warnings)]
```

Comment:
this makes development hard, we should instead have  a ci job which does this, there is a flag that can be passed to rustc to enable this dynamically

Reply:
Ok.